File: app/src/main/java/knf/kuma/tv/anime/RecordPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.pojos.RecordObject;
import knf.kuma.tv.cards.RecordCardView;

public class RecordPresenter extends Presenter {

    public RecordPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new RecordCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((RecordCardView) viewHolder.view).bind((RecordObject) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/RecordPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup

import androidx.leanback.widget.Presenter
import knf.kuma.pojos.RecordObject
import knf.kuma.tv.cards.RecordCardView

class RecordPresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(RecordCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any) {
        (viewHolder.view as RecordCardView).bind(item as RecordObject)
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
